src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -799,6 +798,141 @@ public StringBuilder AppendLine(string value)
             return Append(Environment.NewLine);
         }
 
+        /// <summary>
+        /// Calculate the new length for array allocation when marshalling StringBuilder in interop
+        /// while taking current capacity into account
+        /// This is needed to ensure compat with desktop CLR behavior
+        /// </summary>
+        internal int GetAllocationLength(int requiredLength)
+        {
+            int currentLength = Capacity;
+            if (currentLength < requiredLength)
+            {
+                // round the current length to the nearest multiple of 2
+                // that is >= the required length
+                return (requiredLength + 1) & ~1;
+            }
+
+            return currentLength;
+        }
+
+        /// <summary>
+        /// Throw away the current contents in this StringBuffer and replace it with a new char[]
+        /// NOTE: The buffer in StringBuilder is *not* NULL-terminated.
+        /// This is only called from MCG code
+        /// </summary>
+        internal unsafe void ReplaceBuffer(char* newBuffer)
```

Comment:
This is redundant with:

```
MethodImplAttribute(MethodImplOptions.InternalCall)]
internal unsafe extern void ReplaceBufferInternal(char* newBuffer, int newLength);
```

FCalls, and I expect that the FCall reference will explode once it hits ProjectN.

It may be best to move these interop specific goo to CoreCLR/CoreRT specific files and deal with it separately.